cmd/cluster.go

```diff
@@ -136,3 +149,49 @@ func getClusterVersion(ctx *cli.Context) error {
 	fmt.Printf("Server Version: %s\n", serverVersion)
 	return nil
 }
+
+func clusterUpgradeFromCli(ctx *cli.Context) error {
+	clusterFile, err := resolveClusterFile(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to resolve cluster file: %v", err)
+	}
+	_, _, _, _, err = ClusterUpgrade(clusterFile)
+	return err
+}
+
+func ClusterUpgrade(clusterFile string) (string, string, string, string, error) {
+	logrus.Infof("Upgrading Kubernetes cluster")
+	var APIURL, caCrt, clientCert, clientKey string
+	kubeCluster, err := cluster.ParseConfig(clusterFile)
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, err
+	}
+
+	logrus.Debugf("Getting current cluster")
+	currentCluster, err := kubeCluster.GetClusterState()
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, err
+	}
+	logrus.Debugf("Setting up upgrade tunnels")
+	/*
+		kubeCluster is the cluster.yaml definition. It should have updated configuration
+		currentCluster is the current state fetched from kubernetes
+		we add currentCluster certs to kubeCluster, kubeCluster would have the latest configuration from cluster.yaml and the certs to connect to k8s and apply the upgrade
+	*/
+	kubeCluster.Certificates = currentCluster.Certificates
+	err = kubeCluster.TunnelHosts()
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, err
+	}
+	logrus.Infof("Starting cluster upgrade")
```

Comment:
Can this be a debug info

Reply:
We have a similar one for the `up` command. Added this for consistency.